alphabase/psm_reader/psm_reader.py

```diff
@@ -266,17 +213,41 @@ def set_modification_mapping(
             self.modification_mapping = copy.deepcopy(modification_mapping)
 
         self._str_mods_to_lists()
+
+        if self._add_unimod_to_mod_mapping:
+            self._add_all_unimod()
+            self._extend_mod_brackets()
+
         self.rev_mod_mapping = self._get_reversed_mod_mapping()
 
     def _init_modification_mapping(self) -> None:
         self.modification_mapping = {}
 
+    def _add_all_unimod(self) -> None:
+        for mod_name, unimod in MOD_TO_UNIMOD_DICT.items():
+            if mod_name in self.modification_mapping:
```

Comment:
could we reuse the logic used to add custom mods here? I feel like adding unimod, adding reader specific mods and adding custom mods is very similar.